addons/gdshell/commands/default_commands/ls.gd

```diff
@@ -0,0 +1,119 @@
+extends GDShellCommand
+
+
+signal done_with_instance(instance: Node)
+
+
+var created_an_instance:= false
+
+
+func _main(argv: Array, data) -> Dictionary:
+	done_with_instance.connect(_on_done_with_instance)
+	var path: String
+	
+	# If additional functionality is added, this should be moved to it's own function
+	# But i think i addressed most of what an ls command could do
+	var starting_node = get_starting_node(argv)
+	if starting_node is Dictionary:
+		return starting_node
+	var tree_dict = get_tree_dict(starting_node, starting_node)
+	output_tree_dict(tree_dict)
+	
+	data = tree_dict
+	return {}
+
+
+func get_tree_dict(node: Node, root_node: Node, start:= true) -> Dictionary:
+	root_node = node if start else root_node
+	var node_dict = {}
+	node_dict["name"] = node.name
+	node_dict["has_script"] = node.get_script() != null
+	node_dict["script_file"] = node.get_script().get_path() if node_dict["has_script"] else "none"
+	node_dict["type"] = node.get_class()
+	node_dict["is_instanced_scene"] = node.scene_file_path != ""
+	node_dict["scene_file_path"] = str(node.scene_file_path)
+	node_dict["scene_tree_path"] = str(node.get_path())
+	node_dict["parent"] = node.get_parent().name if node.get_parent() != null else "none"
+	node_dict["children"] = []
+	
+	var child_count := node.get_child_count()
+	
+	for i in child_count:
+		#output(get_tree_dict(node.get_child(i), tree_dict, counter))
+		node_dict["children"].append(get_tree_dict(node.get_child(i), root_node, false))
+
+	if created_an_instance and node.get_parent() == root_node and node.get_index() == root_node.get_child_count() - 1:
+		done_with_instance.emit(root_node)
```

Comment:
This can be just `queue_free()`. I don't see the need for this to be a signal and it is just more straight forward and readable. If it is needed, it can be very easily added back in.
```suggestion
		root_node.queue_free()
```